### PR TITLE
feat: add `WebView::cookies` and `WebView::cookies_for_url` 

### LINKED ISSUE #518: want an API can get webview cookie
**Is your feature request related to a problem? Please describe.**
I want use wry load a remote url, wait user login, and get cookie to use user cookie to call thrid-service api, so I need get cookie on user login sucesss.

**Describe the solution you'd like**

I use electron session to finish it, the sesion can give all webview  cookies, cache, proxy setting...
but I can't found tauri have similar API

**Describe alternatives you've considered**
I found wry can preload script , I try init script and send event to main process, but the remote url will run 302 redict and callback , my init script will loss.

**Would you assign yourself to implement this feature?**
- [ x ] No

**Additional context**
electron session : https://www.electronjs.org/docs/latest/api/session

### PR DESCRIPTION
**Pull request by amrbashir**
closes #518
ref: https://github.com/tauri-apps/tauri/issues/11330

### CODE REVIEW

File: src/wkwebview/mod.rs [hunk 1]
```diff
@@ -825,6 +821,96 @@ r#"Object.defineProperty(window, 'ipc', {
...
+            // cookie is secure and url is https
+            (cookie.secure() && url.scheme() == "https") ||
+            // or cookie is secure and is localhost
+            (
+              cookie.secure() && url.scheme() == "http" && 
+              (url.domain() == Some("localhost") || url.domain().and_then(|d| Ipv4Addr::from_str(d).ok()).map(|ip| ip.is_loopback()).unwrap_or(false))
+            ) ||
+            // or cookie is not secure
+            (!cookie.secure())
+          )
```
Comment (amrbashir) [inline src/wkwebview/mod.rs]: not sure if this is too restrictive, tried to check how webkit2gtk cookies_for_url is implementation but I think their implementation is a bit more relaxed ? https://github.com/WebKit/WebKit/blob/c305a86965236fd8052623faaf8804b1424e7f6b/Source/WebKit/UIProcess/API/APIHTTPCookieStore.cpp#L69

Comment (lucasfernog) [reply src/wkwebview/mod.rs]: when in doubt just ask @tweidinger though I think we should just match the behavior from other platforms if possible

Comment (tweidinger) [reply src/wkwebview/mod.rs]: I think we do match too strict on the path depending on the cookie policy.
For example `/` in the cookie would not match a subpath cookie but it should match any path.

Comment (tweidinger) [reply src/wkwebview/mod.rs]: Just checking the domain match and checking if secure cookie or not sounds acceptable if we don't want to implement the cookie specs fully 😁 .

Comment (amrbashir) [reply src/wkwebview/mod.rs]: alright, pushed a change to match on domain and secure or not. (basically just removed matching the path)

File: src/webkitgtk/mod.rs [hunk 2]
```diff
@@ -804,6 +805,95 @@ impl InnerWebView {
...
+              .map(Self::cookie_from_soup_cookie)
+              .collect()
+          });
+          let _ = tx.send(cookies);
+        })
+      });
+
+    gtk::main_iteration();
+
+    rx.recv()?.map_err(Into::into)
```
Comment (lucasfernog) [inline src/webkitgtk/mod.rs]:
```suggestion
    loop {
      gtk::main_iteration();

      if let Ok(response) = rx.try_recv() {
        return response.map_err(Into::into);
      }
    }
```

File: src/webkitgtk/mod.rs [hunk 3]
```diff
@@ -804,6 +805,95 @@ impl InnerWebView {
...
```
Comment (lucasfernog) [inline src/webkitgtk/mod.rs]:
```suggestion
    loop {
      gtk::main_iteration();

      if let Ok(response) = rx.try_recv() {
        return response.map_err(Into::into);
      }
    }
```

File: .changes/cookies-api.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"wry": "patch:bug"
```
Comment (lucasfernog) [inline .changes/cookies-api.md]:
```suggestion
"wry": "patch"
```

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**